docs/batch_request.md

```diff
@@ -0,0 +1,30 @@
+## BatchRequest
+
+BatchRequest is useful when you want to make multiple requests efficiently. It batches all requests (upto 20 requests) into a json object and makes one api call. You can learn more about it on [Microsoft Docs](https://docs.microsoft.com/en-us/graph/json-batching). 
+
+## Code Sample
+
+```go
+import "github.com/microsoftgraph/msgraph-sdk-go-core"
+import abstractions "github.com/microsoft/kiota-abstractions-go"
+
+reqInfo := client.Me().CreateGetRequestInformation()
+batch := msgraphsdkcore.NewBatchRequest()
+batchItem := batch.AppendBatchItem(*reqInfo)
+
+resp, err := msgraphsdkcore.SendBatch(*batch, reqAdapter)
+
+// print the first response
+fmt.Println(resp.Responses[0].Body)
```

Comment:
additionally the response or the model should be retrieved using the batch item id and not the array index to avoid any kind of confusion, misordering: the service doesn't guarantee the responses will be in the same order as the requests.